contracts/HATVaultsTokenWrapper.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/utils/SafeERC20.sol";
+
+
+/** - user calls approve the vault on tokenwrapper
+- user calls "deposit" on vault -> tokens go from token --> wrapper
+- vault lp = wrapped token
+-------------
+- vault creates lock, calls lpToken.transfer(---)  (wrapper --> wrapper)
+- lock expires --> lock calls lpToken.transfer(lock --> user, hacker, etc)
+
+***/
+
+contract HATVaultsTokenWrapper is ERC20 {
+  using SafeERC20 for IERC20;
+
+  IERC20 public immutable wrappedToken;
+  address public immutable poolContract;
+
+  mapping (address => bool) public stakers;
+  constructor(address _tokenToWrap, address _poolContract, string memory name_, string memory symbol_) ERC20(name_, symbol_) {
+    wrappedToken = IERC20(_tokenToWrap);
+    poolContract = _poolContract;
+  }
+
+  function transferFrom(address _sender, address _recipient, uint256 _amount) public override returns (bool) {
+    // if the recipient is the poolContract, we create new wrapped tokens, and add the sender to the list of stakers
+    // if the recipient is a staker, we unwrap the tokens and send them to the staker
+    // in all other cases, we just use the ERC20 transferFrom
+    if (_recipient == poolContract) {
```

Comment:
and if the safeTransferFrom come from rewardDepositors https://github.com/hats-finance/hats-contracts/blob/develop/contracts/HATVaults.sol#L293 ?  
it is a potential attack vector

Reply:
I do not see how. So that would mean that the caller of the rewardDepositors function is put on the stakers list while he is not a staker. Which is not an attack vector, afaics (mb I should rename the from stakers to `accountsThatKeepTheirTokensUnwrapped`